### PR TITLE
fix: update description of how preimages are revealed with taproot swaps

### PR DESCRIPTION
**Pull request by kilrau**
Preview: https://boltz.gitbook.io/api-boltz-backend-dev/lifecycle

cc @bordalix

### CODE REVIEW

File: docs/lifecycle.md [hunk 1]
```diff
@@ -109,10 +108,11 @@ The following states are traversed in the course of a Reverse Submarine Swap:
...
    confirmation. Boltz broadcasts chain transactions non-RBF only.
-5. `invoice.settled`: the transaction claiming chain bitcoin was broadcast by
-   the user's client and Boltz used the preimage of this transaction to settle
-   the Lightning invoice. This is the final status of a successful Reverse
-   Submarine Swap.
+5. `invoice.settled`: the user's client successfully invoked the API call to
+   construct the claim transaction and supplied the preimage and Boltz then used
+   the preimage to settle the Lightning invoice. This is the final status of a
+   successful Reverse Submarine Swap. Boltz does not track if the client
+   broadcast the claim transaction.
```
Comment (bordalix) [inline docs/lifecycle.md:115]:
```suggestion
   broadcasts the claim transaction.
```

**Review by bordalix — APPROVED** [review]

File: docs/dont-trust-verify.md [hunk 2]
```diff
@@ -92,7 +92,14 @@ match calculated swap amounts to prevent user error.
 Boltz API clients should verify that EVM swaps are carried out using the correct
 contract by comparing the bytecode of the contract.
 
+## Taproot Swaps Script Path Fallback
+
+Key path spends in the Taproot Swap protocol rely on Boltz cooperating. Should
+Boltz disappear or refuse to cooperate, for Chain and Normal Submarine Swaps,
+Boltz API clients must implement a fallback to the script path to be able to
+successfully claim refunds.
```
Comment (michael1011) [inline docs/dont-trust-verify.md:100]:
```suggestion
enforce claim and refund transactions.
```

File: docs/dont-trust-verify.md [hunk 3]
```diff
@@ -92,7 +92,14 @@ match calculated swap amounts to prevent user error.
...
+
 ## General Measurements
 
-Boltz API client should verify that `expectedAmount` in Boltz API responses
-matches locally calculated amount.
+Boltz API clients should calculate swap amounts locally to verify that
+`expectedAmount` in Boltz API responses matches the locally calculated amount.
```
Comment (michael1011) [inline docs/dont-trust-verify.md:105]:
```suggestion
`expectedAmount` in the API responses matches the locally calculated amount.
```

File: docs/lifecycle.md [hunk 4]
```diff
@@ -91,12 +91,11 @@ hash, Boltz creates a hold invoice that can only be settled when the preimage is
...
 doesn't execute, because Boltz doesn't know the preimage yet. Next, Boltz locks
 up chain bitcoin using the same hash so that these can be claimed with the
-previously generated preimage by the client. Once the claim transaction for the
-chain bitcoin is broadcast by the user's Boltz API client, Boltz detects the
-preimage in this transaction and in turn claims its bitcoin by settling the
-Lightning invoice. The [Claim & Refund](claiming-swaps.md) section contains
-details about how Boltz API clients can construct claim transactions for their
-users.
+previously generated preimage by the client. Once the user's client successfully
+invoked the API call to construct the claim transaction and supplied the
```
Comment (michael1011) [inline docs/lifecycle.md:95]: The client constructs; the API just creates a partial signature for it

**Review by michael1011 — CHANGES_REQUESTED** [review]

File: docs/claiming-swaps.md [hunk 5]
```diff
@@ -9,10 +9,9 @@ description: >-
 
 Boltz API clients need to craft and broadcast
 
-- **claim transactions** to claim chain bitcoin and successfully complete
-  **Reverse Submarine Swaps**
-- **refund transactions** to claim refunds for chain bitcoin locked in failed
-  **Submarine Swaps**
+- **claim transactions** for **Reverse Submarine Swaps** & **Chain Swaps**.
+- **refund transactions** to claim refunds for failed **Normal Submarine Swaps**
+  and **Chain Swaps**.
```
[the remark this answers is not in the document]
Comment (kilrau) [reply docs/claiming-swaps.md:14]: Went throught all docs and applied the rule `&` in titles only, in-text `and` only

**Pull request merged**